### ISSUE TITLE
Hot reload YAML prototypes

### ISSUE THREAD

**Issue by snowsignal** [space-wizards/RobustToolbox#1064]
Instead of having to restart the application every single time you make a change to the YAML, what if the game automatically updated whenever you made a change?

### Some ideas on how to do this
- Make this a config option in `server_config.toml`, so that actual servers won't waste time checking for YAML updates
- Every 2 seconds or so (120 ticks if we're doing 60fps), check if any files in the Resources folder have been updated (the simplest way to do this that I can think of would be to hash all the files and compare it with the last hash that was taken)
- Another way to reload files could be through a dedicated console command
- If the files changed, perform a reload of all the prototypes. Changes would be applied by comparing the new and old versions of the prototypes and performing updates on all existing entities in the world (removing components, adding new default components, updating sprites/bounding boxes, etc.)

I plan to work on this once the physics rewrite is finished.

**Comment by SweptWasTaken** [comment]
This would be an insanely convenient feature and would save a *ton* of time on development.

Do you think it could also be down for rsis? Like when a new png is added or a meta.json is edited.

**Comment by snowsignal** [comment]
@SweptWasTaken Yes, I might do that in a separate PR though.

**Comment by SweptWasTaken** [comment]
Doing any of this will gain you my eternal love. Gl.

**Comment by ComicIronic** [comment]
I don't think polling is necessary - each major OS should have a way to set up some kind of file watcher that the OS will then notify whenever something changes.

**Comment by PJB3005** [comment]
.NET provides [`FileSystemWatcher`](https://docs.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher?view=netcore-3.1) for this in a cross-platform manner.

**Comment by DrSmugleaf** [comment]
Done in https://github.com/space-wizards/RobustToolbox/pull/1571
